app/src/main/java/com/mmm/parq/fragments/PictureDialogFragment.java

```diff
@@ -0,0 +1,248 @@
+            @Override
+            public void onResponse(String response) {
+                Gson gson = new Gson();
+                User updatedUser = gson.fromJson(response, User.class);
+                mCallback.setUser(updatedUser);
+
+                Log.d(TAG, "dismissing");
+                dismiss();
+            }
+        }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                Log.e(TAG, "Failed update user request: " + error.getMessage());
+            }
+        }) {
+            @Override
+            protected Map<String, String> getParams() {
+                Map<String, String> params = new HashMap<>();
+                params.put("profilePhotoId", photoId);
+                return params;
+            }
+        };
+
+        RequestQueue queue = HttpClient.getInstance(getActivity().getApplicationContext()).getRequestQueue();
+        queue.add(ratingRequest);
+    }
+
+    // This is jank but s3 doens't like callables so we're doin it.
+    public class UploadCompleteCallback implements Runnable {
+        private String id;
+        public void setId(String id) {
+            this.id = id;
+        }
+
+        @Override
+        public void run() {
+            sendUpdateUserRequest(id + ".jpg");
+        }
+    }
+    private void onImageSelected(Uri uri) {
+        // Store the image in s3
+        S3Manager s3Manager = S3Manager.getInstance(getActivity());
+        FutureTask<Void> userAuthorized = s3Manager.authUser();
+        s3Manager.uploadImage(userAuthorized, uri, new UploadCompleteCallback());
+    }
+
+    private void requestWriteStoragePermission(int requestCode) {
+        if (ContextCompat.checkSelfPermission(getActivity(), Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE},
+                    requestCode);
+        } else {
+            onTakePermissionGranted();
+        }
+    }
+
+    private void requestStoragePermission(int requestCode) {
+        if (ContextCompat.checkSelfPermission(getActivity(), Manifest.permission.READ_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[]{Manifest.permission.READ_EXTERNAL_STORAGE},
+                    requestCode);
+        } else {
+            onStoragePermissionGranted();
+        }
+    }
+
+    private void onTakePermissionGranted() {
+        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        File image = null;
+        try {
+            image = createImageFile();
+        } catch (IOException e) {
+            Log.e(TAG, "Error creating image file: " + e.getMessage());
+        }
+
+        if (image != null) {
+            intent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(image));
+            startActivityForResult(intent, IMAGE_TAKE_REQUEST);
```

Comment:
Yea ok I'm pretty sure that since you're just calling startActivityForResult() here instead of like calling it on the activity or something crazy (I can't remember, but I definitely had an issue with onActivityResult()), you don't have to have the call in the activity up above. I could be wrong though, I'll play with it in a little bit and see if it works.